### ISSUE TITLE
Parameter interface for 0 <= x <= 1

### ISSUE THREAD

**Issue by arose13** [cvxgrp/cvxpy#653]
A convex problem I have is being labeled as non-convex I think because it does not know that the parameter `alpha` is bounded between [0, 1].

I know this from this line fails...
`loss = mse + (1-alpha) * lam * (penalty_1 + penalty_2) + alpha * lam * penalty_3`
 
while this line succeeds...
`loss = mse + lam * (penalty_1 + penalty_2) + lam * penalty_3`

Right now the hyperparameters are parameterized this way. If there is a way to bound them I haven't found it in anywhere in the API
```
alpha = cvx.Parameter(nonneg=True)
alpha.value = 0.5
lam = cvx.Parameter(nonneg=True)
lam.value = 10**(2)
```

**Comment by akshayka** [comment]
CVXPY requires that convex programs are specified using [disciplined convex programming](https://www.cvxpy.org/tutorial/dcp/index.html). Please move this question to [Stack Overflow](https://stackoverflow.com/questions/tagged/cvxpy), and include with your question a complete, executable code snippet that reproduces your problem.
